beacon/handler.go

```diff
@@ -85,6 +83,10 @@ func NewHandler(c net.ProtocolClient, s Store, conf *Config, l log.Logger) (*Han
 		Round:     0,
 	}
 	s.Put(b)
+	// sleep 10ms to make sure at least the first beacon is in the store
+	// on CI server it sometimes happen that the beacon is not even stored
+	// before loading the last beacon
+	time.Sleep(10 * time.Millisecond)
```

Comment:
is there a way to make sure `put` has finished before continued rather than sleeping? e.g piping through a channel to return when it's complete you can wait on or similar?

Reply:
The problem I was trying to solve is that the beacon/chain.go loads the last beacon and it returned a nil beacon instead. My first reaction was that the "put" has not finished. 
But tbh the more I think about it I think "put" returns when it's written in the DB it's a blocking call so it may be something else. It belongs to the some "dirty" tricks I talked to you about. We could try without.

Reply:
delaying with a sleep is often a way of covering up an underlying race / other issue, so just sanity checking

Reply:
I agree - I've removed it - let's run a few times more without this hack.